app/frontend/src/components/Datepicker.test.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  fireEvent,
+  Queries,
+  render,
+  RenderResult,
+  screen,
+  waitFor,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { SUBMIT } from "features/constants";
+import mockdate from "mockdate";
+import { useForm } from "react-hook-form";
+
+import wrapper from "../test/hookWrapper";
+import Datepicker from "./Datepicker";
+
+beforeAll(() => {
+  mockdate.set("2021-03-20");
+});
+
+afterAll(() => {
+  mockdate.reset();
+});
+
+describe("DatePicker", () => {
+  const submitForm = jest.fn();
+  let testWrapper: RenderResult<Queries, HTMLElement>;
+
+  beforeEach(() => {
+    submitForm.mockClear();
+    const Form = () => {
+      const { control, register, handleSubmit } = useForm();
+      const onSubmit = handleSubmit((data) => submitForm(data));
+      return (
+        <form onSubmit={onSubmit}>
+          <Datepicker
+            control={control}
+            error={false}
+            helperText=""
+            id="date-field"
+            inputRef={register}
+            label="Date field"
+            name="datefield"
+          />
+          <input type="submit" name={SUBMIT} />
+        </form>
+      );
+    };
+
+    testWrapper = render(<Form />, { wrapper });
+  });
+
+  it("should submit with proper date for clicking", async () => {
+    await waitFor(() => {
+      fireEvent.click(
+        testWrapper.container.querySelector(".MuiIconButton-label") as Element
+      );
+      fireEvent.click(screen.getByText("23"));
```

Comment:
```suggestion
      userEvent.click(screen.getByText("23"));
```
It's better to use `userEvent` for UI tests (just like what you have below) since it mimics the same events a normal browser would fire.